src/jose/jws/general/signer.ts

```diff
@@ -0,0 +1,59 @@
+import type { GeneralJws, SignatureInput } from './types';
+import type { Signfn } from '../../types';
+
+import { base64url } from 'multiformats/bases/base64';
+import { sign as signEd25519 } from '../../algorithms/ed25519';
+import { sign as signSecp256k1 } from '../../algorithms/secp256k1';
+
+const signers: { [key:string]: Signfn } = {
+  'Ed25519'   : signEd25519,
+  'secp256k1' : signSecp256k1
+};
+
+export class GeneralJwsSigner {
+  private jws: GeneralJws;
+
+  constructor(jws: GeneralJws) {
+    this.jws = jws;
+  }
```

Comment:
Design: similar sentiment to `GeneralJwsVerifier`, not seeing the point of a class, prefer static methods or your common style of exporting function directly, perhaps `sign(...)`.